keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1466 @@
+or Deny rules. For example, intra-tenant traffic management can be delegated to tenant
+admins explicitly with the use of `Pass` rules.
+
+#### Priority
+
+The policy instances will be ordered based on the numeric priority assigned to each
+ANP. `Priority` is a 32 bit integer value, where a smaller number corresponds to
+a higher precedence. The lowest "regular" numeric value (see below for special cases)
+is "1", which corresponds to the highest precedence. Larger numbers have lower precedence.
+For alpha, this API defines "1000" as the maximum numeric value for priority, but
+this may be revisited as the proposal advances. For future-safety, clients may assume
+that higher values will eventually be allowed, and simply treat it as an int32.
+Any positive priority numbered policy will have higher precedence over the namespaced
+NetworkPolicy instances in the cluster, unless the traffic matches a higher-precedence
+`Pass` rule that allows it to bypass any lower-precedence ANP rules.
+
+Additionally, the special priority "0" can be used in the priority field to indicate
+that the rules in that policy instance shall be created at a precedence lower than
+the Namespaced NetworkPolicies. Note that the `Pass` action does not skip ANPs created
+at priority "0". In other words, if some traffic hits a `Pass` rule and no K8s
+NetworkPolicy applies, the "0" priority ANPs will still be evaluated.
+
+The relative precedence of the rules within a single ANP object (all of which
+share a priority) will be determined by the order in which the rule is written.
+Thus, a rule that appears at the top of the ingress/egress rules would take the
+highest precedence. The maximum number of rules, which will be calculated as the
+total summation of the AdminNetworkPolicyIngressRules and AdminNetworkPolicyEgressRules
+in a single ANP instance, will be 100.
+
+Conflict resolution: Two policies are considered to be conflicting if they are assigned
+the same `priority` and apply to the same resources or a union of resources. In order
+to avoid such conflicts, we propose to include tooling for ANP resources to help alert
+the admin to potentially ambiguous ANP priority scenarios, more details in [risks and mitigation](#risks-and-mitigation).
+However, ultimately it will be the job of the network policy implementation to decide
+how to handle overlapping priority situations.
+
+#### Rule Names
+
+In order to help future proof the ANP api, a built in mechanism to identify each
+allow/deny/pass rule is required. Such a mechanism will help administrators organize
+and identify individual rules within an AdminNetworkPolicy resource.
+We propose to introduce a new string field, called `name`, in each `AdminNetworkPolicy`
+ingress/egress rule. Currently the `name` of a rule is optional and does not need to
+be unique, due to the fact that there's no easy way to validate this in-tree. It
+is suggested that implementations provide a mechanism to validate any name
```

Comment:
I don't think we need to call out uniqueness, except to say "name is optional, but it is most useful if it is unique within an ANP instance"

Reply:
Sounds good